### DISCUSSION TITLE
Optimising model on multiple datasets

### DISCUSSION THREAD

**Discussion by arresejo** [optuna/optuna#4388] [Q&A] [answered]
Hello everyone,

I see that in most code examples, the search for the best hyper-parameters of a specific model is done on a single dataset. However, what would be the correct strategy if I wanted to find the best hyper-parameters of a specific model on n different datasets instead of just one?

Kind regards,
Jorge

**Comment by nzw0301** [comment, marked as answer]
I suggest computing an objective value on each dataset and aggregating them into a single objective value, such as a sum of objective values over the datasets. Another approach would be multi-task learning formulation.

**Reply by arresejo** [reply]
I tried the first approach, calculating a specific metric on each dataset and aggregating everything to a mean. It seems to work, thanks a lot for the recommendation :)
